MonstieWash/Assets/Scripts/Erasing/Tasks/TaskData.cs

```diff
@@ -0,0 +1,27 @@
+using UnityEngine;
+
+public class TaskData : MonoBehaviour
+{
+	[SerializeField] protected string m_id;
+
+    [SerializeField] protected float m_progress;
+
+    [SerializeField] protected bool m_complete;
+    [SerializeField] protected float m_threshold;
+
+	[SerializeField] protected Transform m_container;
+
+    public string Id { get => m_id; set => m_id = value; }
+    public float Progress { get => m_progress; set => m_progress = value; }
+    public bool Complete { get => m_complete; set => m_complete = value; }
+    public float Threshold { get => m_threshold; set => m_threshold = value; }
+    public Transform Container { get => m_container;}
+
+	public void OnEnable()
```

Comment:
This is all good, the only thing that will need to change is this line, so that it doesn't reset whenever you move away from the scene.
```suggestion
	public void OnAwake()
```